src/SFML/Graphics/Text.cpp

```diff
@@ -380,6 +380,70 @@ Vector2f Text::findCharacterPos(std::size_t index) const
 }
 
 
+////////////////////////////////////////////////////////////
+bool Text::fitsWithinBounds(const sf::FloatRect& bounds, std::size_t& oob) const
+{
+    // Make sure that we have a valid font
+    if (!m_font)
+    {
+        oob = m_string.getSize();
+
+        return true;
+    }
+
+    // Precompute the variables needed by the algorithm
+    const bool  isBold          = m_style & Bold;
+    float       whitespaceWidth = m_font->getGlyph(U' ', m_characterSize, isBold).advance;
+    const float letterSpacing   = (whitespaceWidth / 3.f) * (m_letterSpacingFactor - 1.f);
+    whitespaceWidth += letterSpacing;
+    const float lineSpacing = m_font->getLineSpacing(m_characterSize) * m_lineSpacingFactor;
+
```

Comment:
Can this code be extracted into a function such this and the code you changed in `findCharacterPos` share a common implementation?

Reply:
Good point, done in 1430fbbe9fc9aa863534a2974d7b9216d3a10620.